Sources/Basics/HTTPClient/HTTPClient+URLSession.swift

```diff
@@ -28,6 +28,19 @@ public struct URLSessionHTTPClient {
         self.downloadTaskManager = DownloadTaskManager(configuration: configuration)
     }
 
+    @available(macOS 12, *)
+    public func execute(
```

Comment:
couple of things:

1. lets add a test that puts some concurrent load on this? there is a similar one already but would be nice to have one that tests the threading across async boundries
2. iirc @neonichu ran into some trouble introducing async APIs since some of the CI / toolchain build nodes did not have the backward support libraries. need to make sure that is addressed so we can finally make use of async APIs in SwiftPM